Client2.py

```diff
@@ -1,45 +1,58 @@
-"""Client Class File"""
+"""This file consists of the client class, its respective methods and their implementations.
+   The client is designed to send, receive data and echo the data simultaneously."""
 
 import socket
+import logging
 
+zero = 0        # Global Variable for zero
+limit = 16      # Data limit
+
+logging.basicConfig(filename='client.log', level=logging.INFO)
 
 class Client:
-    """comment"""
+    """Client class defines the instances and methods required for:
+    1) Creating socket
+    2) Sending Data
+    3) Receiving Data
+    4) Closing the transmission. """
 
     def __init__(self, ip, port):
-        """comment"""
+        """init method defines the necessary instances:
+        1) ip for the ip address or localhost which is used to connect to server.
+        2) Port of connection which is used to connect to the port on which the server is listening.
+        3) Socket object which is used to send and receive the data through.
+        4) Message which has to be sent. """
+
         self.ip = ip
         self.port = port
-        self.Client_Socket = None
+        self.client_socket = None
         self.message = None
-        self.data_received = None
-        self.data_expected = None
-        self.data = None
 
     def create_socket(self):
-        """comment"""
-        self.Client_Socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        """create_socket method creates a socket object and the connects the socket to defined address."""
```

Comment:
Why does create socket method needs to be public?

Reply:
I will include the functionality in init method itself.